### ISSUE TITLE
Nameless "It" blocks don't fail

### ISSUE THREAD

**Issue by beatcracker** [pester/Pester#1744]
## General summary of the issue

You can create a nameless `It` block and Pester wouldn't fail when executing it.

## Describe your environment

Pester version: 5.0.4 C:\pester.5.0.4\Pester.psm1
PowerShell version : 7.0.3
OS version: Microsoft Windows NT 10.0.14393.0

### Example

This code produces 1 successful test result.

```powershell
Describe 'Empty' {
    It {
        throw 'Empty!'
    }
}
```

### Result

```
Invoke-Pester .\Empty.tests.ps1 -Output Detailed

Starting discovery in 1 files.
Discovering in C:\Empty.tests.ps1.
Found 1 tests. 8ms
Discovery finished in 15ms.

Running tests from 'C:\Empty.tests.ps1'
Describing Empty
  [+]
        throw 'Empty!'
     1ms (0ms|1ms)
Tests completed in 101ms
Tests Passed: 1, Failed: 0, Skipped: 0 NotRun: 0
```

## Expected Behavior

Pester should throw in this case, like it does when test name is empty: `It '' {}`

```none
System.Management.Automation.ParameterBindingValidationException: Cannot bind argument to parameter 'Name' because it is an empty string.
```
## Current Behavior

Nameless `It` block produces successful test result.

**Comment by nohwnd** [comment]
What is the benefit of being able to have a test without a name? It would at least confuse nUnit output, unless we set the name to ` <empty>` in that case.

**Comment by beatcracker** [comment]
@nohwnd 

> What is the benefit of being able to have a test without a name?

No benefit at all, on the contrary, I want Pester to throw error in this case. Thing is I accidentally made a nameless `It` block and couldn't figure out why the test code inside it wasn't failing.

**Comment by asears** [comment]
This seems to be a silent failure.  If you pass in "" as the title, it will fail with this message.

```powershell
Starting discovery in 2 files.
System.Management.Automation.ParameterBindingValidationException: Cannot bind argument to parameter 'Name' because it is an empty string.
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.Interpreter.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.LightLambda.RunVoid1[T0](T0 arg0)
   at System.Management.Automation.PSScriptCmdlet.RunClause(Action`1 clause, Object dollarUnderbar, Object inputToProcess)
   at System.Management.Automation.PSScriptCmdlet.DoEndProcessing()
   at System.Management.Automation.CommandProcessorBase.Complete()
at <ScriptBlock>, D:\projects\powershell\pestersandbox\my.silentlysucceeding.Tests.ps1: line 2
at New-Block, C:\Program Files\WindowsPowerShell\Modules\Pester\5.0.4\Pester.psm1: line 691
at Describe, C:\Program Files\WindowsPowerShell\Modules\Pester\5.0.4\Pester.psm1: line 7714
at <ScriptBlock>, D:\projects\powershell\pestersandbox\my.silentlysucceeding.Tests.ps1: line 1
at <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\Pester\5.0.4\Pester.psm1: line 2719
at Invoke-File, C:\Program Files\WindowsPowerShell\Modules\Pester\5.0.4\Pester.psm1: line 2728
at Invoke-BlockContainer, C:\Program Files\WindowsPowerShell\Modules\Pester\5.0.4\Pester.psm1: line 2657
at Discover-Test, C:\Program Files\WindowsPowerShell\Modules\Pester\5.0.4\Pester.psm1: line 1291
at Invoke-Test, C:\Program Files\WindowsPowerShell\Modules\Pester\5.0.4\Pester.psm1: line 2190
at Invoke-Pester<End>, C:\Program Files\WindowsPowerShell\Modules\Pester\5.0.4\Pester.psm1: line 4196
at <ScriptBlock>, <No file>: line 1
```

**Comment by nohwnd** [comment]
@beatcracker ah my bad, I thought you want that to work. Yes that's bad. I think the ScriptBlock is not checking if it is populated, and I think I might have fixed it in the latest 5.1.0 beta, but will double check, and probably port the checks we do in v4.

**Comment by nohwnd** [comment]
This was fixed in 5.1.0-beta1 in this PR. https://github.com/pester/Pester/commit/23a70c294c6724f799493cc52f8cdad4945a9d05#diff-d558fd429a9c73d2d6600557ac2396d05ebb4d9ff4b55db337fa983f776a5868R141

**Comment by iRon7** [comment]
I ran into the same problem as described in this issue.

> What is the benefit of being able to have a test without a name? It would at least confuse nUnit output, unless we set the name to ` <empty>` in that case.

In my case I have a lot of (small) test expressions where any pester error easy identifies what failed:

```PowerShell
Describe 'Test' {
    it '<empty>' { 'one' | should -be 'two' }
}
[-] Test. 20ms (19ms|1ms)
 Expected strings to be the same, but they were different.
 String lengths are both 3.
 Strings differ at index 0.
 Expected: 'two'
 But was:  'one'
 at it '<empty>' { 'one' | should -be 'two' }, C:\Test.ps1:2
 at <ScriptBlock>, C:\Test.ps1:2
Tests completed in 145ms
Tests Passed: 0, Failed: 1, Skipped: 0 NotRun: 0
```

Saying that, if a name is internally required, I would use the ScriptBlock expression itself as an identifier and possibly convert it to Json and truncated this in the display output, something like:

```PowerShell
Describe 'Test' {
    $Test = {
        'one' | 
        should -be 'two'
    }
    It ("$Test" | ConvertTo-Json).SubString(0,31) $Test
}

[-] Test."\r\n        'one' | \r\n       28ms (27ms|1ms)
 Expected strings to be the same, but they were different.
 String lengths are both 3.
 Strings differ at index 0.
 Expected: 'two'
 But was:  'one'
 at should -be 'two', C:\Test.ps1:4
 at <ScriptBlock>, C:\Test.ps1:3
Tests completed in 163ms
Tests Passed: 0, Failed: 1, Skipped: 0 NotRun: 0
```
